docs/_guide/templates.md

```diff
@@ -569,6 +581,162 @@ html`${until(content, html`<span>Loading...</span>`)}`
+
+
+```js
+class MyElement {
+  @queryAll('div')
+  divs;
+
+  render() {
+    return html`
+      <div id="first"></div>
+      <div id="second"></div>
+    `;
+  }
+}
+```
+
+Here, `divs` would return both `<div>` elements in the template. For TypeScript, the typing of a `@queryAll` property is `NodeListOf<HTMLElement>`. If you know exactly what kind of nodes you'll retrieve, the typing can be more specific:
+
+```
+@queryAll('button')
+buttons!: NodeListOf<HTMLButtonElement>
+```
+
+The exclamation point (`!`) after `buttons` is TypeScript's [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator). It tells the compiler to treat `buttons` as always being defined, never `null` or `undefined`.
+
+Finally, `@queryAsync` works like `@query`, except that instead of returning a node directly, it returns a `Promise` that resolves to that node. External code can use this instead of waiting for the `updateComplete` promise. 
+
+This is useful, for example, if the node returned by `@queryAsync` can change as a result of another property change. 
+
+## Accessing slotted children
+
+To access children assigned to slots in your shadow root, you can use the standard `slot.assignedNodes` method and the `slotchange` event.
+
+For example, you can create a getter to access assigned nodes for a particular slot:
+
+```js
+get slottedChildren() {
+  const slot = this.shadowRoot.querySelector('slot');
+  const childNodes = slot.assignedNodes({flatten: true});
+  return Array.prototype.filter.call(childNodes, (node) => node.nodeType == Node.ELEMENT_NODE);
+}
+```
+
+You can also use the `slotchange` event to take action when the assigned nodes change. The following example extracts the text content of all of the slotted children.
+
+```js
+handleSlotchange(e) {
+  const childNodes = e.target.assignedNodes({flatten: true});
+  // ... do something with childNodes ...
+  this.allText = Array.prototype.map.call(childNodes, (node) => {
+    return node.textContent ? node.textContent : ''
+  }).join('');
+}
+
+render() {
+  return html`<slot @slotchange=${this.handleSlotchange}></slot>`;
+}
+```
+
+More information:
+
+*   [HTMLSlotElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement) on MDN.
+
+
+### @queryAssignedNodes decorator
+
+The `@queryAssignedNodes` decorator converts a class property into a getter that returns all of the assigned nodes for a given slot in the component's shadow tree. 
+
+<div class="alert alert-info">
+
+**Using decorators.** Decorators are a proposed JavaScript feature, so you’ll need to use a compiler like Babel or the TypeScript compiler to use decorators. See [Using decorators](decorators) for details.
+
+</div>
+
+```js
+// First argument is the slot name
+// Second argument is `true` to flatten the assigned nodes.
+@queryAssignedNodes('header', true)
+headerNodes;
```

Comment:
Maybe make these private too, though it's a little less of a concern because child nodes are already public external to the element. I just want to reinforce that class members defined only to help with the component implementation should be private by default.

Reply:
Sure, SG.